src/codegen/compile.ml

```diff
@@ -9459,20 +9472,31 @@ and compile_dec env pre_ae how v2en dec : VarEnv.t * G.t * (VarEnv.t -> scope_wr
     assert AllocHow.(match M.find_opt name how with
                      | Some (LocalMut _ | StoreHeap | StoreStatic) -> true
                      | _ -> false);
-      let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
+    let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
 
-      ( pre_ae1,
-        alloc_code,
-        (fun ae -> let sr, code = Var.set_val env ae name in compile_exp_as env ae sr e ^^ code),
-        unmodified
-      )
+    ( pre_ae1,
+      alloc_code,
+      (fun ae -> let sr, code = Var.set_val env ae name in compile_exp_as env ae sr e ^^ code),
+      unmodified
+    )
+  | RefD (name, _, e) ->
+    let pre_ae1, _ = AllocHow.add_local env pre_ae how name in
+
+    ( pre_ae1,
+      G.nop,
+      (fun ae ->
+        let _, prepare_raw, sr, _ = compile_lexp env ae e in
+        prepare_raw ^^
+        Var.capture_aliased_box env ae sr name),
```

Comment:
I am worried about sr showing up here. We are writing a pointer to a local, there cannot be any stackreps, because no values are represented! Probably an artifact of using lexp here, which is not quite right. Maybe just handle aliasing object fields, and write that directly?

Reply:
Yes, what is now in `compile_lexp` will appear here inlined.

Reply:
Does this look better? aa4e131f5